packages/core/src/Serialization/TransactionBody/ProposalProcedure/ParameterChangeAction.ts

```diff
@@ -79,7 +88,16 @@ export class ParameterChangeAction {
 
     const parameterUpdate = ProtocolParamUpdate.fromCbor(HexBlob.fromBytes(reader.readEncodedValue()));
 
-    const action = new ParameterChangeAction(parameterUpdate, govActionId);
+    let policyHash: Hash28ByteBase16 | undefined;
+    if (reader.peekState() === CborReaderState.Null) {
+      reader.readNull();
+    } else {
+      policyHash = HexBlob.fromBytes(reader.readByteString()) as unknown as Hash28ByteBase16;
+    }
+
+    reader.readEndArray();
```

Comment:
This was added, but it seems to be unrelated from the CDDL change.

Could I ask which is its purpose?

Another thing I do not understand is why the same change about `TreasuryWithdrawalsAction` doesn't require this statement.

Thank you

Reply:
I implemented the CBOR reader as a stack-based virtual machine, so every time you read a start array/map/object, you are pushed one level into the stack, you need to read the end of the array/map/object to go out one level on the stack. If you read the contents of an array for example, you need to readStartArray -> read each element of the array -> and finally readEndArray -> then continue reading objects at the same level of the array in the object. 

This is only required if you read more items after the array ends or if there are more elements at the same level of the array (which is not the case here). This readEndArray can probably be skipped, but I think is good to add those because this has the extra benefit of making sure that indefinite arrays are terminated correctly in the CBOR stream.

Reply:
So, if I understood correctly, we could consider a _pseudo problem_ the absence of this statement before this PR; am I right?

And following the same logic, it should be a good idea also adding it at the end of `TreasuryWithdrawalsAction.fromCBOR`: it starts with a `reader.readStartArray()` but I can't find a relative `reader.readEndArray()`; am I missing something else?

Thank you

Reply:
> we could consider a pseudo problem the absence 
Yes, for the sake of setting a good example, we should call `readEndArray()`.